### ISSUE TITLE
complains about missing trusted.d then creates duplicate apt config entry

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#38427]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.86.0-1706698139
- OS Version: Debian/testing

Steps to Reproduce:

1. have previous version already installed
2. migrate the (deprecated) /etc/apt/trusted* key folders to /etc/apt/keyrings/* and rewrite all ".list" files with the new deb822 format ".sources"
3. apt update
4. apt dist-upgrade
5. try again

After step 4 apt complains about a missing directory:
```
Configurazione di code (1.86.0-1706698139)...
/var/lib/dpkg/info/code.postinst: riga 77: /etc/apt/trusted.gpg.d/microsoft.gpg: File o directory non esistente
```

After step 5 it will have created a functional duplicate of /etc/apt/sources.list.d/vscode.sources as vscode.list;
this is somehow related to -189274 in that the root cause is always the same, key difference is that the key itself will apparently not be duplicated and apt won't complain (too much).
